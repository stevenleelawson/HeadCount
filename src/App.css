/*APP COMPONENT RELATED CSS GOES HERE*/
.app {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
