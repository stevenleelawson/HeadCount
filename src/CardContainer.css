.card-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-around;
  margin-top: 1rem;
}
