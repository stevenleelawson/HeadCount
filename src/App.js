import React, { Component } from 'react';
import './App.css';
import KinderData from './data/kindergartners_in_full_day_program';
import DistrictRepository from './helper';
import CardContainer from './CardContainer';
import CompareCard from './CompareCard';
import Header from './Header';


class App extends Component {
  constructor() {
    super();
    this.state = {
      districtArray: [],
      selectedCards: [],
      averages: null
    };

    this.districts = new DistrictRepository(KinderData);
  }

  retrieveData = (userInput = '') => {
    const schools = this.districts.findAllMatches(userInput);
    this.setState({ districtArray: schools});
  }

  handleClick = (event) => {
    const location = event.target.id;
    const selectedCard = this.districts.findByName(location);
    const selectedArray = this.state.selectedCards;
    const alreadyThere = selectedArray.some(district => district.location === location);

    if (alreadyThere) {
      const selectedCards = selectedArray.filter((district) => district.location !== selectedCard.location);
      this.setState({selectedCards});
      return;
    }

    if (selectedArray.length === 0) {
      this.setState({ selectedCards : [selectedCard] });
    } else if (selectedArray.length === 1) {
      this.setState({ selectedCards : [...selectedArray, selectedCard] });
      this.getAverageOfSelected(selectedArray[0], selectedCard);
    } else if (selectedArray.length === 2) {
      const newSelected = [...selectedArray];
      newSelected.shift();
      this.setState({ selectedCards : [...newSelected, selectedCard] });
      this.getAverageOfSelected(newSelected[0], selectedCard);
    }

  }

  getAverageOfSelected = (location1, location2) => {
    const comparison = this.districts.compareDistrictAverages(location1.location, location2.location);
    this.setState({averages:comparison});
  }

  componentDidMount() {
    this.retrieveData();
  }

  render() {
    return (
      <div className='app'>
        <Header filterSchools={this.retrieveData}/>
        {
          this.state.averages &&
          <CompareCard averages={this.state.averages}/>
        }
        <CardContainer schools={this.state.districtArray} handleClick={this.handleClick} toggleClass={this.toggleClass} selected={this.state.selectedCards}/>

      </div>
    );
  }
}

export default App;
