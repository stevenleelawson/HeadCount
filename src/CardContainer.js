import React from 'react';
import Card from './Card';
import './CardContainer.css'

const CardContainer = ({ schools, handleClick, toggleClass, selected }) => {
  const schoolCards = schools.map( (school, index) => {
  const selectedCardLocations = selected.map( name => name.location);
  console.log(selectedCardLocations)
  const selectedClass = selectedCardLocations.includes(school.location) ? 'selected' : ''

    return <Card
      {...school}
      key={school.location}
      handleClick={handleClick}
      toggleClass={toggleClass}
      selectedClass={selectedClass} />
  })
  return (
    <div className="card-container">{schoolCards}</div>
  )
}

export default CardContainer;
