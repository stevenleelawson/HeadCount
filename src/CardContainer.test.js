import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import  CardContainer from './CardContainer';



describe('CardContainer', () => {

  it('should match the snapshot', () => {
    const schools = [
      { location: 'Denver', data: {'2006': .098}},
      { location: 'Golden', data: {'2006': .848}}
    ];

    const wrapper = shallow(<CardContainer schools={schools}/>)
    expect(wrapper).toMatchSnapshot();
  });

});
