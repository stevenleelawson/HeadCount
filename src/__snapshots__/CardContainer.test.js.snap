// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    handleClick={[Function]}
    schools={
        Array [
            Object {
              "data": Object {
                "2006": 0.098,
              },
              "location": "Denver",
            },
            Object {
              "data": Object {
                "2006": 0.848,
              },
              "location": "Golden",
            },
          ]
    }
    selected={
        Array [
            Object {
              "data": Object {
                "2006": 0.098,
              },
              "location": "Denver",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          data={
                    Object {
                              "2006": 0.098,
                            }
          }
          handleClick={[Function]}
          location="Denver"
          selectedClass="selected"
/>,
        <Card
          data={
                    Object {
                              "2006": 0.848,
                            }
          }
          handleClick={[Function]}
          location="Golden"
          selectedClass=""
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Denver",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "2006": 0.098,
          },
          "handleClick": [Function],
          "location": "Denver",
          "selectedClass": "selected",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Golden",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "2006": 0.848,
          },
          "handleClick": [Function],
          "location": "Golden",
          "selectedClass": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            data={
                        Object {
                                    "2006": 0.098,
                                  }
            }
            handleClick={[Function]}
            location="Denver"
            selectedClass="selected"
/>,
          <Card
            data={
                        Object {
                                    "2006": 0.848,
                                  }
            }
            handleClick={[Function]}
            location="Golden"
            selectedClass=""
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Denver",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "2006": 0.098,
            },
            "handleClick": [Function],
            "location": "Denver",
            "selectedClass": "selected",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Golden",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "2006": 0.848,
            },
            "handleClick": [Function],
            "location": "Golden",
            "selectedClass": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
